### PR TITLE
Rework the mechanism for code sharing a little

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/EndpointsArtifactTask.java [hunk 1]
```diff
@@ -110,10 +116,62 @@ public void setBasePath(String basePath) {
...
+      getProject().mkdir(outputDirectory);
+    }
+
+    List<String> params = new ArrayList<>();
+
+    params.add(command);
+
+    params.add("-o");
+    if (!Strings.isNullOrEmpty(outputFileName)) {
+      params.add(new File(outputDirectory, outputFileName).getAbsolutePath());
```
Comment (coollog) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/EndpointsArtifactTask.java:170]: This is using `getAbsolutePath` whereas the original was using `getPath`. I guess this is safer?

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/EndpointsArtifactTask.java [hunk 2]
```diff
@@ -123,26 +181,24 @@ void generateEndpointsArtifact() throws Exception {
...
-    }
-
-    List<String> params =
-        new ArrayList<>(endpointsTaskConfiguration.getActionSpecificParams(outputDirectory));
-
     params.add("-cp");
     params.add(classpath);
 
     params.add("-w");
     params.add(webAppDir.getPath());
```
Comment (coollog) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/EndpointsArtifactTask.java]: This looks like it's still using just `getPath` though.

**Review by coollog — APPROVED** [review]
More red than green looks so nice

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/EndpointsArtifactTask.java (hunk 1 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/task/EndpointsArtifactTask.java:170]: Yeah I can't remember why I would have used getPath initially (I'm used to using getAbsolutePath), I'll change the -w to use getAbsolutePath too.

**Pull request merged**